lib/glimesh/api/resolvers/account_resolver.ex

```diff
@@ -29,6 +30,90 @@ defmodule Glimesh.Api.AccountResolver do
     end
   end
 
+  def follow_channel(
+        _parent,
+        %{channel_id: channel_id, live_notifications: live_notifications},
+        %{context: %{access: access}}
+      ) do
+    with :ok <- Bodyguard.permit(Glimesh.Api.Scopes, :follow, access),
+         channel when channel != nil <- ChannelLookups.get(channel_id, [:user]),
+         streamer when streamer != nil <- Accounts.get_user(channel.user.id),
+         user when user != nil <- Accounts.get_user(access.user.id),
```

Comment:
You can convert these into pattern match guarantees, something like:
```suggestion
         %Channel{} = channel <- ChannelLookups.get(channel_id, [:user]),
         %User{} = streamer <- Accounts.get_user(channel.user.id),
         %User{} = user <- Accounts.get_user(access.user.id),
```

Reply:
Perfect! Thanks I didn't learn this syntax from reading the pattern matching docs, updated the code and it's looking pretty clean at this point I think.